File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/VerificationAdapter.java
package com.workforcex.employer.ui.employer;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.workforcex.employer.R;
import com.workforcex.employer.api.Verification;
import java.util.List;

public class VerificationAdapter extends RecyclerView.Adapter<VerificationAdapter.VH> {

    private final List<Verification> items;
    private final OnVerificationAction onAction;

    public interface OnVerificationAction {
        void onUpdate(String verificationId, String status, String comments);
    }

    public VerificationAdapter(List<Verification> items, OnVerificationAction onAction) {
        this.items = items;
        this.onAction = onAction;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_verification, parent, false);
        return new VH(v);
    }

    @Override
    public void onBindViewHolder(@NonNull VH h, int pos) {
        Verification item = items.get(pos);
        h.tvWorkerName.setText("Worker: " + item.user.mobileNumber);
        h.tvVerificationType.setText("Type: " + item.verificationType);

        h.btnApprove.setOnClickListener(v -> onAction.onUpdate(item.id, "VERIFIED", ""));
        h.btnReject.setOnClickListener(v -> onAction.onUpdate(item.id, "REJECTED", ""));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class VH extends RecyclerView.ViewHolder {
        TextView tvWorkerName, tvVerificationType;
        Button btnApprove, btnReject;

        VH(View v) {
            super(v);
            tvWorkerName = v.findViewById(R.id.tvWorkerName);
            tvVerificationType = v.findViewById(R.id.tvVerificationType);
            btnApprove = v.findViewById(R.id.btnApprove);
            btnReject = v.findViewById(R.id.btnReject);
        }
    }
}

File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/VerificationAdapter.kt
package com.workforcex.employer.ui.employer

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.workforcex.employer.R
import com.workforcex.shared_employer.models.Verification

class VerificationAdapter(
    private val items: List<Verification>
) : RecyclerView.Adapter<VerificationAdapter.VH>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_verification, parent, false)
        return VH(v)
    }

    override fun onBindViewHolder(h: VH, pos: Int) {
        val item = items[pos]
        h.tvWorkerName.text = "Worker: " + (item.user.name ?: item.user.mobileNumber)
        h.tvVerificationType.text = "Type: " + item.verificationType

        h.btnView.setOnClickListener {
            val context = h.itemView.context
            val intent = Intent(context, VerifyWorkerActivity::class.java).apply {
                putExtra("workerId", item.user.id)
                putExtra("workerName", item.user.name)
            }
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int = items.size

    class VH(v: View) : RecyclerView.ViewHolder(v) {
        val tvWorkerName: TextView = v.findViewById(R.id.tvWorkerName)
        val tvVerificationType: TextView = v.findViewById(R.id.tvVerificationType)
        val btnView: Button = v.findViewById(R.id.btnView)
    }
}
